server/login.cpp

```diff
@@ -1112,10 +1149,21 @@ bool Create::getDeity(const std::shared_ptr<Socket>& sock, std::string str, int
     const RaceData* race = gConfig->getRace(sock->getPlayer()->getRace());
     if(mode == Create::doPrint) {
 
+
+        if(sock->getPlayer()->getClass() == CreatureClass::CLERIC)
+            sock->print("Note: Clerics of Linothan must be male. Clerics of Mara must be female.\n\n");
         sock->print("Please choose a deity:\n");
 
         for(l=1, k=0 ; l<DEITY_COUNT ; l++) {
             if(race->allowedDeity(sock->getPlayer()->getClass(), sock->getPlayer()->getSecondClass(), l)) {
+                
+                //Gender restrict for Mara, Arachnus, Linothan clerics
+                if( sock->getPlayer()->getClass() == CreatureClass::CLERIC &&
+                    ((sock->getPlayer()->flagIsSet(P_MALE) && l == MARA) ||
```

Comment:
This looks like it allows male Maras and disallows male Linothans

Reply:
It is disallowing what is incorrect is what it's doing. If your male it will not provide the Mara option. Same thing with the Linothan one. Arachnus has to be female too for when they go in. It works extensively tested it.